### PR TITLE
[SPARK-11550][Docs] Replace example code in mllib-optimization.md using include_example

### PR DESCRIPTION
**Pull request by pravingadakh**

### CODE REVIEW

File: examples/src/main/scala/org/apache/spark/examples/mllib/LBFGSExample.scala [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+import org.apache.spark.mllib.util.MLUtils
+// $example off$
+
+import org.apache.spark.{SparkConf, SparkContext}
+
+object LBFGSExample {
+
+  def main(args: Array[String]) {
+
+    val conf = new SparkConf().setAppName("SimpleFPGrowth")
```
Comment (yinxusen) [inline examples/src/main/scala/org/apache/spark/examples/mllib/LBFGSExample.scala]: rename the appname

File: examples/src/main/scala/org/apache/spark/examples/mllib/LBFGSExample.scala [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+import org.apache.spark.mllib.linalg.Vectors
+import org.apache.spark.mllib.optimization.{LBFGS, LogisticGradient, SquaredL2Updater}
+import org.apache.spark.mllib.util.MLUtils
+// $example off$
+
+import org.apache.spark.{SparkConf, SparkContext}
+
+object LBFGSExample {
+
+  def main(args: Array[String]) {
```
Comment (yinxusen) [inline examples/src/main/scala/org/apache/spark/examples/mllib/LBFGSExample.scala]: add `:Unit = ` as the return value

File: examples/src/main/scala/org/apache/spark/examples/mllib/LBFGSExample.scala [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    // Get evaluation metrics.
+    val metrics = new BinaryClassificationMetrics(scoreAndLabels)
+    val auROC = metrics.areaUnderROC()
+
+    println("Loss of each step in training process")
+    loss.foreach(println)
+    println("Area under ROC = " + auROC)
+    // $example off$
+  }
+
```
Comment (yinxusen) [inline examples/src/main/scala/org/apache/spark/examples/mllib/LBFGSExample.scala]: remove the blank line

File: examples/src/main/java/org/apache/spark/examples/mllib/JavaLBFGSExample.java [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+import org.apache.spark.mllib.classification.LogisticRegressionModel;
+import org.apache.spark.mllib.evaluation.BinaryClassificationMetrics;
+import org.apache.spark.mllib.linalg.Vector;
+import org.apache.spark.mllib.linalg.Vectors;
+import org.apache.spark.mllib.optimization.*;
+import org.apache.spark.mllib.regression.LabeledPoint;
+import org.apache.spark.mllib.util.MLUtils;
+import org.apache.spark.SparkConf;
+import org.apache.spark.SparkContext;
+
```
Comment (yinxusen) [inline examples/src/main/java/org/apache/spark/examples/mllib/JavaLBFGSExample.java]: Here should have a `$example off$`.

File: examples/src/main/java/org/apache/spark/examples/mllib/JavaLBFGSExample.java [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+public class JavaLBFGSExample {
+  public static void main(String[] args) {
+    SparkConf conf = new SparkConf().setAppName("L-BFGS Example");
+    SparkContext sc = new SparkContext(conf);
```
Comment (yinxusen) [inline examples/src/main/java/org/apache/spark/examples/mllib/JavaLBFGSExample.java]: Here should have a `$example on$`

File: examples/src/main/java/org/apache/spark/examples/mllib/JavaLBFGSExample.java [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+      new BinaryClassificationMetrics(scoreAndLabels.rdd());
+    double auROC = metrics.areaUnderROC();
+
+    System.out.println("Loss of each step in training process");
+    for (double l : loss)
+      System.out.println(l);
+    System.out.println("Area under ROC = " + auROC);
+  }
+}
+// $example off$
```
Comment (yinxusen) [inline examples/src/main/java/org/apache/spark/examples/mllib/JavaLBFGSExample.java]: remove this line.

File: examples/src/main/java/org/apache/spark/examples/mllib/JavaLBFGSExample.java [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+
+    // Get evaluation metrics.
+    BinaryClassificationMetrics metrics =
+      new BinaryClassificationMetrics(scoreAndLabels.rdd());
+    double auROC = metrics.areaUnderROC();
+
+    System.out.println("Loss of each step in training process");
+    for (double l : loss)
+      System.out.println(l);
+    System.out.println("Area under ROC = " + auROC);
```
Comment (yinxusen) [inline examples/src/main/java/org/apache/spark/examples/mllib/JavaLBFGSExample.java]: `$example off$` here

Comment (yinxusen) [inline examples/src/main/java/org/apache/spark/examples/mllib/JavaLBFGSExample.java]: `$example off$` here

**Comment by yinxusen** [comment]
@pravingadakh Everything looks good except for the positions of comments.

**Comment by pravingadakh** [comment]
@yinxusen I have added example on off labels at places you mentioned.

**Comment by yinxusen** [comment]
@mengxr LGTM

**Comment by mengxr** [comment]
ok to test

**Comment by mengxr** [comment]
Merged into master and branch-1.6. Thanks!

**Pull request closed without merging**